src/platform/packages/shared/kbn-esql-utils/src/utils/append_to_query.test.ts

```diff
@@ -58,10 +58,10 @@ describe('appendToQuery', () => {
       );
     });
 
-    it('appends a where clause in an existing query with casting to string when the type is not given', () => {
-      expect(appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '-')).toBe(
+    it('appends a where clause in an existing query inferring whether to cast when the type is not given', () => {
```

Comment:
Why dont we create a new test case and we overwrite this one? This should work as expected

Reply:
This does not work the same with the type inference, because if it is a `string` type already, it won't try to explicitly cast to a string (it's redundant). The explicit casting the value to string when encountering an unmapped field was the faulty behaviour, because unmapped fields are not always string types or are compatible with strings.

There may be other edge cases relating to this though, so further test cases to find these will probably be a good idea (though we'd need to figure out just what those cases are for unmapped fields). I figure this change might be a bit controversial, but the existing behaviour is buggy without a doubt.

Reply:
Can we do not cast in string only the unmapped fields or do not allow to append in query for these? I am afraid that changing the current behavior will create bugs.